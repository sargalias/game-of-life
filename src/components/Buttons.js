import React from 'react'
import Button from './Button';

const Buttons = (props) => {
  let text;
  let onClick;
  if (props.isRunning) {
    text = 'Pause';
    onClick = props.onPauseClick;
  } else {
    text = 'Run';
    onClick = props.onRunClick;
  }
  return (
    <div className="button-container">
      <Button text={text} onClick={onClick} />
      <Button text="Clear" onClick={props.onClearClick}/>
      <Button text="Reset" onClick={props.onResetClick}/>
    </div>
  );
};

export default Buttons;
