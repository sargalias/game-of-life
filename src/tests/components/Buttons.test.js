import React from 'react';
import Buttons from '../../components/Buttons';
import { shallow } from 'enzyme';

test('should render Buttons correctly without props', () => {
  const wrapper = shallow(<Buttons />);
  expect(wrapper).toMatchSnapshot();
});

/*
test('should render Buttons correctly with props', () => {
  const wrapper = shallow(<Buttons
    onRunClick={'a'}
    onPauseClick={'b'}
    onClearClick={'c'}
    onResetClick={'d'}
  />);
  expect(wrapper).toMatchSnapshot();
});
*/

test('should render buttons correctly with prop.isRunning=true', () => {
  const wrapper = shallow(<Buttons
    onRunClick={'a'}
    onPauseClick={'b'}
    onClearClick={'c'}
    onResetClick={'d'}
    isRunning={true}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('should render buttons correctly with prop.isRunning=false', () => {
  const wrapper = shallow(<Buttons
    onRunClick={'a'}
    onPauseClick={'b'}
    onClearClick={'c'}
    onResetClick={'d'}
    isRunning={false}
  />);
  expect(wrapper).toMatchSnapshot();
});

