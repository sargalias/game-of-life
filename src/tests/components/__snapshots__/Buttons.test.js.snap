// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Buttons correctly without props 1`] = `
<div
  className="button-container"
>
  <Button
    text="Run"
  />
  <Button
    text="Clear"
  />
  <Button
    text="Reset"
  />
</div>
`;

exports[`should render buttons correctly with prop.isRunning=false 1`] = `
<div
  className="button-container"
>
  <Button
    onClick="a"
    text="Run"
  />
  <Button
    onClick="c"
    text="Clear"
  />
  <Button
    onClick="d"
    text="Reset"
  />
</div>
`;

exports[`should render buttons correctly with prop.isRunning=true 1`] = `
<div
  className="button-container"
>
  <Button
    onClick="b"
    text="Pause"
  />
  <Button
    onClick="c"
    text="Clear"
  />
  <Button
    onClick="d"
    text="Reset"
  />
</div>
`;
